src/osx/cocoa/dnd.mm

```diff
@@ -314,20 +326,35 @@ - (NSDragOperation)code
     return resultCode;
 }
 
-- (NSDragOperation)draggingSourceOperationMaskForLocal:(BOOL)forLocal
-{
-    /*
-    By default drag targets receive a mask of NSDragOperationAll (0xf)
-    which, despite its name, does not include the later added
-    NSDragOperationMove (0x10) that sometimes is wanted.
-    Use NSDragOperationEvery instead because it includes all flags.
+/*
+By default drag targets receive a mask of NSDragOperationAll (0xf)
+which, despite its name, does not include the later added
+NSDragOperationMove (0x10) that sometimes is wanted.
+Use NSDragOperationEvery instead because it includes all flags.
+
+Note that this, compared to the previous behaviour, adds
+NSDragOperationDelete to the mask which seems harmless.
+
+We are also keeping NSDragOperationLink and NSDragOperationPrivate
+in to preserve previous behaviour.
+*/
+#if wxOSX_USE_DRAG_SESSION
+
+- (NSDragOperation)draggingSession:(nonnull NSDraggingSession *)session sourceOperationMaskForDraggingContext:(NSDraggingContext)context {
+    NSDragOperation allowedDragOperations = NSDragOperationEvery;
+
+    if (m_dragFlags == wxDrag_CopyOnly)
+    {
+        allowedDragOperations &= ~NSDragOperationMove;
```

Comment:
IIUC the comment above, I think we should also clear the `NSDragOperationDelete` bit here.

Reply:
@vadz interesting question, we don't have a match for a 'delete drag' to the trash in wx don't we ? actually it seems like a reasonable addition. How is this handled on other platforms ?

Reply:
I've added an update that does not allow a drag to the trash anymore. That way we don't give the appearance of supporting a drag-delete

Reply:
FWIW I don't think other platforms have "delete" drag and while I didn't check it, it would be logical to suppose that dragging to the trash would use "move".